nautobot/core/utils/migrations.py

```diff
@@ -0,0 +1,35 @@
+import sys
+from django.db import models
+
+
+def check_for_duplicates_with_natural_key_fields_in_migration(model_class, natural_key_fields):
+    """
+    Migration helper method to raise a RuntimeError if the existing model_class data contains incorrigible duplicate records.
+
+    Args:
+        model_class: Nautobot model class (Device, VirtualChassis and etc)
+        natural_key_fields: Names of the fields serving as natural keys for the model_class.
+    """
+    duplicate_records = (
+        model_class.objects.values(*natural_key_fields)
+        .order_by()
+        .annotate(count=models.Count(natural_key_fields))
+        .filter(count__gt=1)
+    )
+    print("\n    Checking for duplicate records ...")
+    if duplicate_records.exists():
+        if len(natural_key_fields) > 1:
+            print(
+                f"    Duplicate {model_class.__name__} attributes '{*natural_key_fields,}' detected: {list(duplicate_records.values_list(*natural_key_fields))}",
```

Comment:
How does `{*natural_key_fields,}` get rendered? Not familiar with that syntax.

Reply:
It's something like this:
```
Python 3.10.11 (main, May 23 2023, 14:03:36) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> test_list = ["name", "address", "SSN"]
>>> f_string = f"render the list {*test_list,}"
>>> print(f_string)
render the list ('name', 'address', 'SSN')
>>> 
```

Reply:
Thanks, got it. So it's casting the list to a tuple and rendering that.